source/dubregistry/api.d

```diff
@@ -24,13 +24,32 @@ void registerDubRegistryAPI(URLRouter router, DubRegistry registry)
 {
 	auto pkgs = new LocalDubRegistryAPI(registry);
 	router.registerRestInterface(pkgs, "/api");
+	router.get("/api/packages/dump", delegate(req, res) => dumpPackages(req, res, registry));
 }
 
 /// Compatibility alias.
 deprecated("Use registerDubRegistryAPI instead.")
 alias registerDubRegistryWebApi = registerDubRegistryAPI;
 
 
+private void dumpPackages(HTTPServerRequest req, HTTPServerResponse res, DubRegistry registry)
+{
+	import vibe.data.json : serializeToPrettyJson;
+	import vibe.stream.wrapper : StreamOutputRange;
+
+	res.contentType = "application/json; charset=UTF-8";
+	res.headers["Content-Encoding"] = "gzip"; // force GZIP compressed response
+	auto dst = StreamOutputRange(res.bodyWriter);
+	dst.put('[');
+	bool first = true;
+	foreach (p; registry.getPackageDump()) {
+		if (!first) dst.put(',');
+		else first = false;
+		serializeToPrettyJson(&dst, p);
```

Comment:
> But in a future revision, letting the client control this with an "Accept" header would be an option.

Yeah I was just curious ;-)
If this becomes a bottleneck, there are a lot of other strategies (incremental, caching, ...) that we could try as well which could potentially save more time / resources, so l am not advocating to do "premature optimization" here.